pyvista/core/utilities/points.py

```diff
@@ -459,3 +464,91 @@ def vector_poly_data(orig, vec):
+
+    The axes are computed as the eigenvectors of the covariance matrix from the
+    mean-centered points, and are processed to ensure that they form a right-handed
+    coordinate frame.
+
+    The axes explain the total variance of the points. The first axis explains the
+    largest percentage of variance, followed by the second axis, followed again by
+    the third axis which explains the smallest percentage of variance. The axes may
+    be used to build an oriented bounding box or to align the points to another set
+    of axes (e.g. the world XYZ axes).
+
+    .. note::
+        The computed axes are not unique, and the sign of each axis direction can be
+        arbitrarily changed.
+
+    .. versionadded:: 0.45.0
+
+    Parameters
+    ----------
+    points : MatrixLike[float]
+        Nx3 array of points.
+
+    return_sizes : bool, default: False
+        If ``True``, also returns the axes sizes. The sizes are computed as the square
+        root of the eigen vectors of the mean-centered covariance matrix.
+
+    Returns
+    -------
+    numpy.ndarray
+        3x3 orthonormal array with the principal axes as row vectors.
+
+    numpy.ndarray
+        Three-item array of the axes sizes.
+
+    Examples
+    --------
+    Create a mesh with points that have the largest variation in ``X``,
+    followed by ``Y``, then ``Z``.
+
+    >>> import pyvista as pv
+    >>> mesh = pv.ParametricEllipsoid(xradius=10, yradius=5, zradius=1)
+    >>> p = pv.Plotter()
+    >>> _ = p.add_mesh(mesh)
+    >>> _ = p.show_grid()
+    >>> p.show()
+
+    Compute its principal axes.
+
+    >>> principal_axes = pv.principal_axes(mesh.points)
+    >>> principal_axes
+    array([[-1.0000000e+00,  1.6606515e-10,  1.9055901e-10],
+           [-1.6606515e-10, -1.0000000e+00, -5.0257620e-10],
+           [ 1.9055901e-10, -5.0257620e-10,  1.0000000e+00]], dtype=float32)
+
+    Note that the principal axes have ones along the diagonal and zeros
+    in the off diagonals. This indicates that the first principal axis is
+    aligned with the x-axis, the second with the y-axis, and third with
+    the z-axis. This is expected, since the mesh is already axis-aligned.
+
+    However, since the signs of the principal axes are arbitrary, the
+    first and third axes in this case have a negative direction.
+
+    """
+    points = _validation.validate_arrayNx3(points)
+
+    points_centered = points - np.mean(points, axis=0)
+    eig_vals, eig_vectors = np.linalg.eigh(points_centered.T @ points_centered)
+    sizes = np.sqrt(eig_vals)[::-1]  # ascending order -> descending order
+    axes = eig_vectors.T[::-1]  # columns, ascending order -> rows, descending order
+
+    # Normalize and ensure output forms right-handed coordinate frame
+    axes[0] /= np.linalg.norm(axes[0])
+    axes[1] /= np.linalg.norm(axes[1])
+    axes[2] = np.cross(axes[0], axes[1])
+    if type(axes) is not np.ndarray:
+        axes = np.array(axes, dtype=axes.dtype)
```

Comment:
Why not
```suggestion
    axes = np.asarray(axes)
```
?

Reply:
Yup way better, I even went looking at the numpy docs to find this.

But, I think we can probably remove this altogether, and return `pyvista_ndarray` if we have `pyvista_ndarray` in as suggested elsewhere.